core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariDecommissionTimeCalculator.java

```diff
@@ -0,0 +1,53 @@
+package com.sequenceiq.cloudbreak.service.cluster.ambari;
+
+import static com.sequenceiq.cloudbreak.api.model.Status.AVAILABLE;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.core.flow2.stack.FlowMessageService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.Msg;
+import com.sequenceiq.cloudbreak.domain.Template;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.host.HostMetadata;
+
+@Component
+public class AmbariDecommissionTimeCalculator {
+
+    private static final int DECOMMISSIONING_MAGIC_NUMBER = 12;
+
+    private static final int SECONDS_PER_MINUTE = 60;
+
+    @Inject
+    private FlowMessageService flowMessageService;
+
+    public void calculateDecommissioningTime(Stack stack, Collection<HostMetadata> filteredHostList, Map<String, Map<Long, Long>> dfsSpace, long usedSpace) {
+        Optional<HostMetadata> aHostMetadata = filteredHostList.stream().findFirst();
+        if (aHostMetadata.isPresent()) {
+            Template template = aHostMetadata.get().getHostGroup().getConstraint().getInstanceGroup().getTemplate();
+            long eachNodeCapacityInGB = template.getVolumeCount() * template.getVolumeSize() + template.getRootVolumeSize();
+            long usedGlobalDfsSpace = dfsSpace.values().stream()
+                    .mapToLong(dfsSpaceByUsed -> dfsSpaceByUsed.values().stream().mapToLong(Long::longValue).sum())
+                    .sum();
+            long remainingGlobalDfsSpace = dfsSpace.values().stream()
+                    .mapToLong(dfsSpaceByUsed -> dfsSpaceByUsed.keySet().stream().mapToLong(Long::longValue).sum())
+                    .sum();
+            double clusterUtilizationFraction = usedGlobalDfsSpace / (double) (usedGlobalDfsSpace + remainingGlobalDfsSpace);
+            double decommissionFraction = usedSpace / (double) (usedGlobalDfsSpace + remainingGlobalDfsSpace);
+            double decommissionSeconds = DECOMMISSIONING_MAGIC_NUMBER * eachNodeCapacityInGB * clusterUtilizationFraction * decommissionFraction;
+            long decommissionMinutes = Math.round(decommissionSeconds / SECONDS_PER_MINUTE);
+            String timeString = String.valueOf(decommissionMinutes) + " minutes";
```

Comment:
Would be better to move to else branch to avoid unnecessary string concatenation